### PR TITLE
chore: enable tracing of function timing for debug subcommands

### PR DESCRIPTION
**Pull request by ReactorScram**
e.g. `#[tracing::instrument]` will now print the time a function spent busy and idle without having to add a timer crate or manually capture `Instant`s

Split off from #5357 

Doesn't affect any production code

### CODE REVIEW

File: rust/headless-client/src/lib.rs [hunk 1]
```diff
@@ -581,8 +584,10 @@ pub fn debug_command_setup() -> Result<()> {
     let filter = EnvFilter::builder()
         .with_default_directive(tracing_subscriber::filter::LevelFilter::INFO.into())
         .from_env_lossy();
-    let layer = fmt::layer().with_filter(filter);
-    let subscriber = Registry::default().with(layer);
+    let subscriber = SubscriberBuilder::default()
+        .with_span_events(FmtSpan::CLOSE)
```
Comment (conectado) [inline rust/headless-client/src/lib.rs:588]: This will log for every span close? this sounds like it can be a bit too noisy by default

Comment (conectado) [reply rust/headless-client/src/lib.rs:588]: I mean for debugging

Comment (ReactorScram) [reply rust/headless-client/src/lib.rs:588]: Yeah, as soon as I switch to debug log level, it logs every single `decapsulate` call.

What if we use this instead? https://crates.io/crates/logging_timer
It's a new dependency but then I can opt-in individual functions to timing

Comment (conectado) [reply rust/headless-client/src/lib.rs:588]: do we need it in main? I figure we will only want this very sparsely when debugging performance

Comment (ReactorScram) [reply rust/headless-client/src/lib.rs:588]: Yeah I'll see if I can get by without it. Hopefully it's easy to find again next time. That also means we wouldn't have even manual regression tests for things like startup time

**Comment by ReactorScram** [comment]
Closing, I ended up using `logging_timer` to do this a different way, in #5375

**Pull request closed without merging**